libsmtutil/SMTLib2Interface.cpp

```diff
@@ -110,6 +112,41 @@ void SMTLib2Interface::addAssertion(Expression const& _expr)
 	m_commands.assertion(toSExpr(_expr));
 }
 
+namespace
+{
+std::vector<std::string> parseValuesFromResponse(std::string const& _response)
+{
+	std::stringstream ss(_response);
+	std::string answer;
+	ss >> answer;
+	smtAssert(answer == "sat");
+
+	std::vector<SMTLib2Expression> parsedOutput;
+	SMTLib2Parser parser(ss);
+	try
+	{
+		while (!parser.isEOF())
+			parsedOutput.push_back(parser.parseExpression());
+	}
+	catch(SMTLib2Parser::ParsingException&)
+	{
+		return {};
+	}
+	smtAssert(parsedOutput.size() == 1, "SMT: Expected model values as a single s-expression");
+	auto const& values = parsedOutput[0];
+	smtAssert(!isAtom(values));
```

Comment:
I think you're overusing these asserts a bit, although to be frank, it does make it a bit easier to read. `asSubexpressions()` and `asAtom` will already assert that the passed expression is not an atom.

Reply:
You are right, in both points.
I think it still conveys meaningful message about what is expected to be true in the caller.